src/main/scala/akka/persistence/jdbc/journal/dao/ByteArrayJournalDao.scala

```diff
@@ -49,6 +50,16 @@ trait BaseByteArrayJournalDao extends JournalDaoWithUpdates {
   import journalConfig.daoConfig.{ batchSize, bufferSize, logicalDelete, parallelism }
   import profile.api._
 
+  val logger = LoggerFactory.getLogger(this.getClass)
+
+  // This logging may block since we don't control how the user will configure logback
+  // We can't use a Akka logging neither because we don't have an ActorSystem in scope and
+  // we should not introduce another dependency here.
+  // Therefore, we make sure we only log a warning for logical deletes once
+  lazy val logWarnAboutLogicalDeletionDeprecation = {
+    logger.warn("Logical deletion of events is deprecated and will be removed in version 4.0.0")
```

Comment:
"will be removed in akka-persistence-jdbc version 4.0.0"

Also it might be a good idea to include the name of the config value the user needs to change to disable this.